armi/reactor/components/component.py

```diff
@@ -1217,7 +1217,8 @@ def getReactionRateDict(nucName, lib, xsType, mgFlux, nDens):
     Parameters
     ----------
     nucName - str
-        nuclide name -- e.g. 'U235'
```

Comment:
Thanks for this update @keckler! 

Here are a couple comments:

1. Let's change the `xsType` input to be `microSuffix`, so that this encapsulates both the XS Type (i.e., A, B, C) as well as the burnup group (i.e., A, B, C) so that we can actually get reaction rates based on burnup-dependent cross sections.

To do this, we can use `self.getMicroSuffix()`, where on the block level this is implemented as is, but on a component this is implemented as reaching up to its parent and getting it from the block-level.

2. It would be advantageous for us to standardize the dictionary of reaction rates that can be obtained from the nuclide data library - `{"nG": 0, "nF": 0, "n2n": 0, "nA": 0, "nP": 0, "n3n": 0}`. This is a bit of a refactor, but having this mapping of these strings to the following xsLibrary micros would be good:

```
        ("nG", libNuc.micros.nGamma),
        ("nF", libNuc.micros.fission),
        ("n2n", libNuc.micros.n2n),
        ("nA", libNuc.micros.nalph),
        ("nP", libNuc.micros.np),
```

Just grepping around the code base, I see this mapping occurring in `blocks.py/getReactionRates`, `composites.py/getReactionRates`, `component.py/getReactionRateDict`,  `isotopicsDepletionInterface.py/makeXsecTable`, and `crossSectionTable.py/CrossSectionTable`. 

It might make sense to have the strings for `nG`, `nF`, `n2n`, `nA`, `nP`, etc. mapped back to `xsCollections.py` module-level constants that already exist:

```
NGAMMA = "nGamma"      # radiative capture
NAPLHA = "nalph"       # (n, alpha)
NP = "np"              # (n, proton)
ND = "nd"              # (n, deuteron)
NT = "nt"              # (n, triton)
FISSION_XS = "fission" # (n, fission)
N2N_XS = "n2n"         # (n,2n)
NUSIGF = "nuSigF"      
NU = "neutronsPerFission"
```

Reply:
I agree with all the things you suggest above, but can we possibly make that into a separate PR?

I think technically what I implemented "fixed" the bug, and the things you're suggesting are feature requests.

Just trying to limit the scope of this PR so that I can get it into the codebase more quickly.

Reply:
Sure thing! Ill make an issue.